src/main/java/de/tum/cit/aet/artemis/communication/web/ForwardedMessageResource.java

```diff
@@ -0,0 +1,104 @@
+import de.tum.cit.aet.artemis.communication.repository.ForwardedMessageRepository;
+import de.tum.cit.aet.artemis.core.exception.BadRequestAlertException;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastStudent;
+import de.tum.cit.aet.artemis.core.util.TimeLogUtil;
+
+/**
+ * REST controller for managing ForwardedMessages.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping(value = "api/")
+public class ForwardedMessageResource {
+
+    private static final Logger log = LoggerFactory.getLogger(ForwardedMessageResource.class);
+
+    private static final String ENTITY_NAME = "forwardedMessage";
+
+    private final ForwardedMessageRepository forwardedMessageRepository;
+
+    public ForwardedMessageResource(ForwardedMessageRepository forwardedMessageRepository) {
+        this.forwardedMessageRepository = forwardedMessageRepository;
+    }
+
+    /**
+     * POST /forwarded-messages : Create a new forwarded message.
+     *
+     * @param forwardedMessage the forwarded message to create.
+     * @return the ResponseEntity with status 201 (Created) and with the body containing the new forwarded message,
+     *         or with status 400 (Bad Request) if the forwarded message already has an ID.
+     * @throws BadRequestAlertException if the forwarded message already has an ID.
+     */
+    @PostMapping("forwarded-messages")
+    @EnforceAtLeastStudent
+    public ResponseEntity<ForwardedMessageDTO> createForwardedMessage(@RequestBody ForwardedMessage forwardedMessage) throws URISyntaxException {
+        if (forwardedMessage.getId() != null) {
+            throw new BadRequestAlertException("A new forwarded message cannot already have an ID", ENTITY_NAME, "idExists");
+        }
+        ForwardedMessage savedMessage = forwardedMessageRepository.save(forwardedMessage);
+        ForwardedMessageDTO dto = new ForwardedMessageDTO(savedMessage);
+        return ResponseEntity.created(new URI("/api/forwarded-messages/" + savedMessage.getId())).body(dto);
+    }
+
+    /**
+     * GET /forwarded-messages : Retrieve forwarded messages grouped by their destination IDs.
+     *
+     * @param ids  a set of destination IDs (either post or answer IDs) for which forwarded messages should be retrieved.
+     * @param type the type of destination ('post' or 'answer') to specify whether the IDs belong to posts or answers.
+     * @return the ResponseEntity with status 200 (OK) and a list of ForwardedMessagesGroupDTO objects,
+     *         where each object contains a destination ID and the associated forwarded messages.
+     * @throws BadRequestAlertException if the type parameter is invalid or unsupported.
+     */
+    @GetMapping("forwarded-messages")
+    @EnforceAtLeastStudent
+    public ResponseEntity<List<ForwardedMessagesGroupDTO>> getForwardedMessages(@RequestParam Set<Long> ids, @RequestParam String type) {
+
+        log.debug("GET getForwardedMessages invoked with ids {} and type {}", ids, type);
```

Comment:
looks like security checks (student in course) and student has access to those forwarded messages is missing